utils/tls_config.go

```diff
@@ -0,0 +1,125 @@
+package utils
+
+import (
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	"github.com/docker/notary/trustmanager"
+)
+
+// Client TLS cipher suites (dropping CBC ciphers for client preferred suite set)
+var clientCipherSuites = []uint16{
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+}
+
+// Server TLS cipher suites
+var serverCipherSuites = append(clientCipherSuites, []uint16{
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+	tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+	tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+}...)
+
+// ServerTLSOpts generates a tls configuration for servers using the
+// provided parameters.
+type ServerTLSOpts struct {
+	ServerCertFile    string
+	ServerKeyFile     string
+	RequireClientAuth bool
+	ClientCADirectory string
```

Comment:
Is there any actual (or planned) need for having a directory here? The only user so far are tests.

This makes using a single file awkward (as the tests already manifest), and the asymmetry between ServerTLSOpts and ClientTLSOpts is awkward.

grpc/credentials.NewClientTLSFromFile is using x509.CertPool.AppendCertsFromPEM, loading multiple CAs from a single file; this would support the same functionality and be easier to use in the simple case.